src/lib.rs

```diff
@@ -66,6 +69,15 @@ impl ToString for SdpConnection {
     }
 }
 
+impl AnonymizingClone for SdpConnection {
+    fn masked_clone(&self, anon: &mut StatefulSdpAnonymizer) -> Self {
+        let mut masked = self.clone();
+        masked.addr = anon.mask_ip(&self.addr);
```

Comment:
I'm not sure I like that this first clone's the whole thing, to then free the cloned addr and replace it with the masked.
Wouldn't it make more sense to invoke the SdpConnection constructor instead? The arguments for the constructor would be direct clones of self attributes which don't require masking. And invoke the masking functions on the ones which require masking.
Just trying to reduce the cloning here.